File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
import sys
input = sys.stdin.readline

N = int(input())

dp = [5001 for _ in range(N+1)]

if N == 3 or N == 5:
    print(1)
elif N == 4:
    print(-1)
else:
    dp[3] = 1
    dp[5] = 1

    for i in range(6, N+1):
        dp[i] = min(dp[i-3], dp[i-5]) + 1

    if dp[-1] < 5000:
        print(dp[-1])
    else:
        print(-1)

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {

	static int N;
	
	public static void main(String[] args) throws Exception{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		N = Integer.parseInt(br.readLine());
		
		int[] dp = new int[5005];
		for (int i=0; i<dp.length; i++) {
			dp[i] = 5005;
		}
		
		dp[3] = 1;
		dp[5] = 1;
		
		for (int i = 6; i<dp.length; i++) {
			dp[i] = Math.min(dp[i-3], dp[i-5]) + 1;
		}
		
		if (dp[N] > 5000) System.out.println(-1);
		else System.out.println(dp[N]);
	}

}